### ISSUE TITLE
Better error message for expired PGP key.

### ISSUE THREAD

**Issue by yishilin14** [mssun/passforios#284]
I am using v0.7.0(2) now. This issue first appears in v0.6.0(20). While using an expired PGP key, the error message is not informative.

- When creating a new entry, the error message is "EncryptionError".
- When saving an edited entry, the encryption step fails silently and nothing is saved.

**Comment by yishilin14** [comment]
@ZortacDev Do you have any ideas about how to provide more informative error messages about encryption?

**Comment by ZortacDev** [comment]
Yeah, sure. My Go code receives the errors but fails silently at the moment because it's a bit of a hassle to pass the multiple return values of the Go functions to Swift code. The GopenPGP guys are reworking their API at the moment though, which should make my glue code obsolete. I'll take a look tomorrow and (depending on the state of the GopenPGP API) start reworking my code.

**Comment by yishilin14** [comment]
> When saving an edited entry, the encryption step fails silently and nothing is saved.

This part is (sort of) fixed in  d6271499484141c9b7ceb4a2694d18fe7383f244. Now, one will also see "EncryptionError" after clicking "save".

**Comment by mssun** [comment]
@ZortacDev, are you talking about this PR on refactoring API (https://github.com/ProtonMail/gopenpgp/pull/6). I saw that it is merged, but I'm not sure about the stability.

**Comment by ZortacDev** [comment]
Yeah, that’s probably it. I used the API tagged as v1.0.0 (which is older than the v0 API). I just noticed that the documentation had changed quite a bit while working on my code and decided to use the older version as the new version had some compilation issues with iOS (those issues appear to be resolved with ProtonMail/gopenpgp#17). If everything works I’ll port my code over to the new API (maybe even get rid of the glue code and use gopenpgp directly) and implement proper error handling. If it doesn’t work, I’ll implement the error handling but keep the glue code as-is otherwise. I’ve got a bit of time on my hands in the next few days so this should get done rather quickly.

If we’re really worried about stability, we could ditch gopenpgp completely and use some lower level OpenPGP crypto provided by the Go extended standard library (golang.org/x/crypto/openpgp). I’d have to do some further reading though to get that working.


On 15. July 2019 at 06:45:39, Mingshen Sun (notifications@github.com) wrote:

@ZortacDev, are you talking about this PR on refactoring API (ProtonMail/gopenpgp#6). I saw that it is merged, but I'm not sure about the stability.
